Library/Homebrew/cask/cask.rb

```diff
@@ -349,6 +350,14 @@ def api_to_local_hash(hash)
       hash
     end
 
+    def ruby_source_checksum
+      return JSON.parse(@source)["ruby_source_checksum"] if loaded_from_api
+
+      {
+        "sha256" => Digest::SHA256.file(sourcefile_path).hexdigest,
+      }
+    end
```

Comment:
I'd rather not have to parse the source twice here.

```suggestion
    def ruby_source_checksum
      @ruby_source_checksum ||= {
        "sha256" => Digest::SHA256.file(sourcefile_path).hexdigest,
      }
    end
```

You could do this instead and just pass it as another parameter to the constructor when it's loaded from the API though the list of values being passed to the constructor just keeps growing.

Reply:
The amount of parameters we pass to the constructor is a mess but it probably is the better option for now.